server/src/routers/MenuItemsRouter/parseMenuItemData.js

```diff
@@ -0,0 +1,53 @@
+import parseNutritionInfo from './parseNutritionInfo';
+import filters from '../../config/filters.js';
+
+/**
+ * Parses and combines the nutrition (by call to parseNutritionInfo), 
+ * filter, and ingredient data into a single json menu
+ *
+ * throws an error if any of the inputs are null or empty 
+ * or missing DATA field or empty DATA field
+ *
+ * @param {dict} nutritionData
+ * @param {dict} filterData
+ * @param {dict} ingredientData
+ *
+ * return {dict} with menu as a single json object
+ *
+ */
+
+export default function parseMenuItemData(nutritionData, filterData, ingredientData) {
+	// Error checking for invalid/empty data object
+	if (!nutritionData || !nutritionData.DATA || !nutritionData.DATA.length || !nutritionData.DATA[0].length) {
+    	throw new Error('Empty nutrition object returned from YaleDining API');
+    } 
+    if(!filterData || !filterData.DATA || !filterData.DATA.length || !filterData.DATA[0].length){
+    	throw new Error('Empty filter object returned from YaleDining API');
+    } 
+    if(!ingredientData || !ingredientData.DATA || !ingredientData.DATA.length || !ingredientData.DATA[0].length){
+    	throw new Error('Empty ingredient object returned from YaleDining API');
+    }
+
+    // Process ingredient list
+	var ingredientList = [];
+	for (var i = 0; i < ingredientData.DATA.length; i++){
+		ingredientList.push(ingredientData.DATA[i][1]);
+	}
+	
+
+	// Process filters 
+	var filterList = []; // list of applicable filters
+	for(var i = 0; i < filters.length; i++){
+		if(filterData.DATA[0][i+2] == 1) {
+			filterList.push(filters[i]);
+		}
+	}
+    return {
+    	name : nutritionData.DATA[0][1], 
+    	nutrition: parseNutritionInfo(nutritionData.DATA[0]), 
+    	ingredients: ingredientList,
+    	filterProperties: filterList,
+    	rating: 5 // this line is temporarily hard coded
```

Comment:
Nit, would help if you added the string TODO to this comment so we can find it easily in the future

Reply:
Done